tests/functional/syntax/test_constants.py

```diff
@@ -171,6 +171,13 @@ def hello() :
     """,
         ImmutableViolation,
     ),
+    (
+        # hex values not allowed for signed integer types
```

Comment:
might be ok to allow this actually

Reply:
Done!

Reply:
hm if we allow hex literals for unsigned and signed ints, should we disallow folding involving any hex integer? since we do not know at the time of folding if a hex value should be interpreted as a signed integer or an unsigned integer

Reply:
but I think excluding hex literals during folding would not work either because it would break things downstream where we expect a literal value during typechecking

Reply:
what if we make the syntax more explicit like native hex bytes? e,g, unsigned hex integers would be `u"0xff"` and signed hex integers would be `i"0xff"`.